lib/cn/vblock_reader.h

```diff
@@ -52,9 +52,9 @@ struct vbr_madvise_work {
  * from media and the relevant information is stored in a @vblock_desc struct.
  */
 struct vblock_desc {
-    struct kvs_mblk_desc vbd_mblkdesc; /* underlying block descriptor */
-    uint32_t             vbd_off;      /* byte offset of vblock data */
-    uint32_t             vbd_len;      /* byte length of vblock data */
+    const struct kvs_mblk_desc *vbd_mblkdesc; /* underlying block descriptor */
+    uint32_t             vbd_off;      /* byte offset of vblock data (always 0!) */
+    uint32_t             vbd_len;      /* byte length of vblock data (not including footer) */
```

Comment:
The other descriptors (kblock, hblock) have the `kvs_mblk_desc` embedded.  This descriptor has a ptr to the `kvs_mblk_desc` because for vblocks, the mblock desc lives in the mbset.